uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -523,7 +533,12 @@ mod filters {
     pub(super) fn type_name(
         as_ct: &impl AsCodeType,
         ci: &ComponentInterface,
+        config: &Config,
     ) -> Result<String, askama::Error> {
+        if let Some(overwrite_type_name) = config.rename.get(&as_ct.as_codetype().type_label(ci)) {
+            return Ok(overwrite_type_name.to_owned());
+        }
```

Comment:
If you moved the renaming to happen in KotlinCodeOracle then you wouldn't need to do anything special here, since `type_label` ends up calling `KotlinCodeOracle::class_name`.

This code makes me realize you'd need to change `type_label` to input a `Config`, but I think that should be fine.